dags/common/enhance.py

```diff
@@ -0,0 +1,54 @@
+import datetime
+
+
+class Enhancement:
+    def __init__(self):
+        self.creation_date = datetime.datetime.now().isoformat()
+
+    def constuct_abstracts(self, item, publisher):
+        item["abstracts"] = [{"value": item.pop("abstracts", ""), "source": publisher}]
+
+    def construct_acquisition_source(self, item, publisher):
+        item["acquisition_source"] = {
+            "source": publisher,
+            "method": publisher,
+            "date": self.creation_date,
+            "submission_number": "path/to/the/file",
+        }
+
+    def construct_copyright(self, item):
+        item["copyright"] = [
+            {
+                "holder": item.pop("copyright_holder", ""),
+                "year": item.pop("copyright_year", ""),
+                "statement": item.pop("copyright_statement", ""),
+                "material": item.pop("copyright_material", ""),
+            }
+        ]
+
+    def construct_imprints(self, item, publisher):
+        item["imprints"] = [
+            {"date": item.pop("date_published", ""), "publisher": publisher}
+        ]
+
+    def construct_record_creation_date(self, item):
+        item["record_creation_date"] = self.creation_date
+
+    def construct_titles(self, item, publisher):
+        item["titles"] = [
+            {
+                "title": item.pop("title", ""),
+                "subtitle": item.pop("subtitle", ""),
+                "source": publisher,
+            }
+        ]
+
+    def __call__(self, publisher, item):
```

Comment:
I'm not a fan of using the __call__ function to enhance. I would define a function called `enhance`

Reply:
Good proposal. For me **call** fits better here, since the class does just one job- enhance a given dict.  There cannot be any more actions done with that

Reply:
Yes I get your point. I still don't really like it but it's in domain of personal preference here so nothing to change 👍